### PR TITLE
(CDAP-14591) Re-add the set of RecordFormat classes

### PR DESCRIPTION
**Pull request by chtyim**
- Refactored to not use StreamEvent

### CODE REVIEW

**Comment by chtyim** [comment]
This is a revert of the commit for #10946 for a set of cdap-formats classes, with refactoring to not use StreamEvent. They are still used by some of the hydrator plugins.

File: cdap-api-common/src/main/java/co/cask/cdap/api/data/format/Formats.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.data.format;
+
+/**
+ * Names of {@link RecordFormat RecordFormats} built into the platform.
+ */
+public final class Formats {
```
Comment (sreevatsanraman) [inline cdap-api-common/src/main/java/co/cask/cdap/api/data/format/Formats.java:22]: can we keep track of this breaking change to update docs

Comment (chtyim) [reply cdap-api-common/src/main/java/co/cask/cdap/api/data/format/Formats.java:22]: All these are part of incompatible change due to removal of Stream.

**Comment by chtyim** [comment]
Build passed https://builds.cask.co/browse/CDAP-DUT6725

**Comment by albertshau** [comment]
Do we know which ones still use it? https://github.com/cdapio/hydrator-plugins/pull/881 removes it from hydrator-plugins and it would be nice to not have avro and grok dependencies, as they are explicitly excluded by several of the projects that use this module.

**Comment by chtyim** [comment]
@albertshau it is the Kafka streaming plugin.

**Comment by chtyim** [comment]
@albertshau See https://builds.cask.co/download/CDAP-BUT-RCU/build_logs/CDAP-BUT-RCU-1630.log

Both the Kafka streaming and FileStreamingSource is using StreamEvent

**Review by albertshau — APPROVED** [review]
lgtm

**Pull request merged**